src/main/java/io/jenkins/plugins/artifact_manager_jclouds/JCloudsArtifactManager.java

```diff
@@ -121,7 +121,7 @@ private String getBlobPath(String key, String path) {
     public void archive(FilePath workspace, Launcher launcher, BuildListener listener, Map<String, String> artifacts)
             throws IOException, InterruptedException {
         LOGGER.log(Level.FINE, "Archiving from {0}: {1}", new Object[] { workspace, artifacts });
-        Map<String, String> contentTypes = workspace.act(new ContentTypeGuesser(new ArrayList<>(artifacts.keySet()), listener));
+        Map<String, String> contentTypes = workspace.act(new ContentTypeGuesser(new ArrayList<>(artifacts.values()), listener));
```

Comment:
I would like to create a test for this (probably with a custom `Step`), but I need AWS access.

Reply:
> I need AWS access.

If you are just amending #303, you should be able to extend the self-contained MinIO test. That may or may not reproduce the `MalformedSecurityHeader` error, but you could at least assert the uploaded content type and so confirm the fix.

> with a custom `Step`

Yeah https://github.com/jenkinsci/workflow-step-api-plugin/blob/master/src/test/java/org/jenkinsci/plugins/workflow/steps/StepExecutions.java#L56 should be easiest.

Reply:
> If you are just amending https://github.com/jenkinsci/artifact-manager-s3-plugin/pull/303, you should be able to extend the self-contained MinIO test

I think https://github.com/jenkinsci/artifact-manager-s3-plugin/pull/107 introduced the issue, but ok I will look into `MinIO`.

Reply:
Never mind, https://github.com/jenkinsci/artifact-manager-s3-plugin/blob/c9b867f4a2844f97212faed95ea35bc45baf1ae6/src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/JCloudsArtifactManagerTest.java#L281-L307 is not part of https://github.com/jenkinsci/artifact-manager-s3-plugin/blob/c9b867f4a2844f97212faed95ea35bc45baf1ae6/src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/MinioIntegrationTest.java#L56. Possibly it could be.

Reply:
https://github.com/jenkinsci/artifact-manager-s3-plugin/pull/349/commits/4311349edd6518abfc265d885f02e4b3bab877bc introduces an (untested) test.